docs/current_docs/quickstart/agent/part2.mdx

```diff
@@ -0,0 +1,377 @@
+
+    ```java file=./snippets/part3/agent/java/agent.java.snippet
+    ```
+
+    Be sure to add imports for the following:
+
+    ```java
+import io.dagger.client.Secret;
+import io.dagger.client.GithubIssue;
+    ```
+    </TabItem>
+</Tabs>
+
+The `develop-issue` function connects the dots in the automation flow. It will be called when a Github issue is ready to be developed. Using the `github-issue` dependency, this code will:
+
+- Read the title and body from the Github issue
+- Use the body as the assignment for the agent
+- Open a pull request with the Directory returned by the agent
+
+### Create a Github Actions workflow
+
+Github Actions will need the ability to call out to an LLM. It will need the same environment variables you [configured](../../configuration/llm.mdx) locally setup as repository secrets.
+
+To configure new repository secrets:
+- From your Github repository, select the Settings tab
+- Under Security, expand the section "Secrets and Variables" and select Actions
+- Select "New repository secret"
+- Create a secret for each variable required for your LLM provider. For example: `GEMINI_API_KEY`
+
+![Configure Github Actions repository secrets](/img/current_docs/quickstart/agent/configure_gha_llm.png)
+
+If you signed up for Dagger Cloud, you should also configure a `DAGGER_CLOUD_TOKEN` repository secret to see your Github Actions traces in Dagger Cloud.
+
+To find your Dagger Cloud token, navigate to the organization settings page using the cogwheel icon in the top navigation bar. Under the `Tokens` sub-menu, click the eye icon to view the token. You can also use this URL pattern: `https://v3.dagger.cloud/{Your Org Name}/settings?tab=Tokens`
+
+![Get token](/img/current_docs/configuration/get-token.png)
+
+The repository also needs to be configured to give Github Actions the ability to create pull requests. On the left side, select Actions -> General and find the option "Allow GitHub Actions to create and approve pull requests". Make sure this is enabled.
+
+![Allow Github Actions to create pull requests](/img/current_docs/quickstart/agent/configure_gha_prs.png)
+
+Create the file `.github/workflows/develop.yml` with the following content:
+
+```yaml file=./snippets/part3/develop.yml
+```
+
+This snippet shows an example with Google Gemini by specifying `GEMINI_API_KEY`. Change this to match the repository secrets you created earlier.
+
+This Github Actions workflow will call the `develop-issue` function when the label "develop" is added to an issue. The Github token passed to the agent has permissions to read issues, write contents, and write pull-requests.
+
+Commit and push your changes to Github:
+
+```shell
+git add .
+git commit -m'deploy agent to github'
+git push
+```
+
+Create a new Github issue in your repository. You can title it "change the page color" with the description "make the main page green".
```

Comment:
```suggestion
Create a new Github issue in your repository. Title it "change the page color" with the description "make the main page green".
```

Maybe this should be a different color, to distinguish it from the previous run?

Reply:
Sure, changing the previous run so that I don't need to take new screenshots